static/app/components/globalSelectionLink.tsx

```diff
@@ -41,41 +42,41 @@ class GlobalSelectionLink extends React.Component<Props> {
       typeof to === 'object' && to.query ? {...globalQuery, ...to.query} : globalQuery;
 
     if (location) {
-      let toWithGlobalQuery;
+      let toWithGlobalQuery: LocationDescriptor;
```

Comment:
Did this to make sure everything was still sane with the typings

Reply:
```suggestion
         const toWithGlobalQuery: LocationDescriptor = hasGlobalQuery
        ? typeof to === 'string'
          ? {pathname: to, query}
          : {...to, query}
        : {};
```

or 

```suggestion
      let toWithGlobalQuery: LocationDescriptor = {};
```

🤷‍♀️

Reply:
sgtm, I like that it get's rid of the `!`

Reply:
yes that was the idea :)